utility/README.md

```diff
@@ -1,679 +1,1309 @@
+```mermaid
+---
+title: Signature Validation By Servicer
+---
+stateDiagram-v2
+    state "Get gateways the App<br>delegated to: [G1, G2]" as getGateways
+    state "Is relay signed by one of:<br>Application, Gateway1, Gateway2?" as sigCheck
+
+    state "Valid (should service relay)" as Valid
+    state "Invalid (do not service relay)" as Invalid
 
+    [*] --> getGateways
+    getGateways --> sigCheck
 
+    sigCheck --> Valid: Yes
+    sigCheck --> Invalid: No
 ```
-# ValidatorPause Interface
-type ValidatorPauseMsg interface {
-  GetAddress()  Address       # The address of the Validator being paused
-}
 
-# ValidatorUnpause Interface
-type ValidatorUnpauseMsg interface {
-  GetAddress()  Address       # The address of the Validator being unpaused
-}
+Servicer's are incentivized to respond to any valid relay since it is applicable for reward distribution. The session tokens used for rate limiting by the Servicer will come out of the same bucket as described in the [rate limiting algorithm](#315-rate-limiting) regardless of who in the ring signed the request.
+
+```mermaid
+sequenceDiagram
+    actor A as Application
+    participant WS as World State
+    actor G as Gateway
+    actor S as Servicer
+
+    A ->>+ G: StartSession
+    G ->> G: StartSession
+    G ->>- A: AccessToken
+
+    loop Session Duration
+        A ->> +G: Request(AccessToken)
+        G ->> G: Sign Request
+        G ->> +S: SignedRequest
+        S ->> S: Validate Signature<br>& Session Limits
+        S ->> S: Handle Request<br>& Sign Response
+        S ->> -G: SignedResponse
+        G ->> G: ** Gateway specific features ** <br>(altruist, check, challenge, proof, etc...)
+        G ->> -A: Response
+    end
 ```
 
+#### 3.5.6 Registration
 
-A Validator is able to modify their initial staking values including the ServiceURL, OperatorPubKey, and StakeAmount by submitting a StakeMsg while already staked. It is important to note, that a StakeAmount is only able to be modified greater than or equal to the current value. This allows the protocol to not have to track pieces of stake from any Validator and enables an overall simpler implementation. 
+Registration differs from staking in the sense that the pubKey is known but there are no economic benefits/penalties in this stage of the protocol's progression.
```

Comment:
somewhat related to economic incentives, but we should consider refractoring the entire the [“Block rewards](https://github.com/pokt-network/pocket-network-protocol/tree/issues/23/aat/utility#362-block-rewards) to apply for all actors in the network in a generic form, so that way it is not inclusive to just DAO rewards and becomes a generic formula based off governance parameters. Example use case: Incentivizing gateway actors to run gateways and earn a section of rewards they help serve. After all, gateway actors have to “bond” some of their POKT through security deposits so i can expect some on chain incentivized reward to accompany it. I guess it can always come later, with the cost of a chain upgrade coordination.